bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/util/TestUtil.java

```diff
@@ -0,0 +1,37 @@
+package com.google.cloud.bigtable.util;
+
+import com.google.cloud.bigtable.config.Logger;
+import io.grpc.Server;
+import io.grpc.ServerBuilder;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+
+public class TestUtil {
+  private static final Logger LOG = new Logger(TestUtil.class);
+
+  public static Server buildServer(ServerBuilderFactory serverBuilderFactory) throws IOException {
+    int maxRetries = 3;
+
+    for (int retry = 0; retry < maxRetries; retry++) {
+      int port;
+      try (ServerSocket s = new ServerSocket(0)) {
+        port = s.getLocalPort();
+      }
+      Server server = serverBuilderFactory.buildServer(port).build();
```

Comment:
I wanted the interface to return the server builder so that we can call the start in here and simplify the clien code.
I'm open to having the ServerBuilder.forPort(port) call be in here too

Reply:
FWIW, we have something similar in the veneer client: https://github.com/googleapis/java-bigtable/blob/master/google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/FakeServiceHelper.java

Reply:
Cool, good to have a similar reference. I'd suggest changing that to have retries too, but I don't _think_ that codebase is used in our integration tests so the flakes are less prominent (though I think it would be fine to add still)